web/shared/src/typing_status.ts

```diff
@@ -12,7 +12,11 @@ export type Recipient =
       }
     | (StreamTopic & {
           message_type: "stream";
-      });
+      })
+    | {
+          message_type: "single_message";
```

Comment:
`message_type` = `single_message` might be a weird name for this.... I thought of it since typing notifications for edit messages are more like dealing with individual messages than a `stream` or `direct` type.

Reply:
I think probably you should just add a separate field, maybe called `notification_event_type` field that can be `typing` or `typing_message_edit` (or whatever we picked in API design).

It's best to use the same terms across the code base for the same idea, since that makes `git grep` nice, and so a new `single_message` type is just confusing.